src/port1.0/portutil.tcl

```diff
@@ -664,6 +664,10 @@ proc variant {args} {
 proc variant_isset {name} {
     global variations
 
+    if {$name eq "universal" && [llength [option configure.universal_archs]] < 2} {
+        return 0
+    }
+
```

Comment:
> Hm, this fakes that it is not selected, even if it was?

Yes.

> Shouldn't we stop the selection and not the query whether it is selected?

"selection" happens when the user writes `+universal` on the command line or in variants.conf. How do you propose to prevent the user from doing that? It could also occur if the portfile author wrote `default_variants +universal`, and here again I'm trying to avoid having to modify each portfile that does so.

Reply:
The variant can only be selected if the port provides it. I thought we were discussing to get rid of that? With this change, the `port info` output would still show "universal" (correct me if I am wrong), but specifying `+universal` has no effect. This would be confusing.

Reply:
> The variant can only be selected if the port provides it.

No, that is not correct. As I explained in [ticket #57133](https://trac.macports.org/ticket/57133), "`variant_isset` has the perhaps somewhat unexpected behavior that a variant can be set even if that variant is not defined".

> With this change, the `port info` output would still show "universal" (correct me if I am wrong)

With *only* this change to `proc variant_isset`, that would be correct. But this change was not intended to be used alone. See the other proposed change in this PR—making `proc variant` not create the variant if the name is "universal" and there are not at least 2 `universal_archs`. Unfortunately that change causes problems described earlier in this PR which I don't know how to resolve.